Common/Util/PythonUtil.cs

```diff
@@ -25,27 +25,109 @@ namespace QuantConnect.Util
+                return method(pyObject, typeof(T1), typeof(T2)).AsManagedObject(typeof(Action<T1, T2>));
+            }
+        }
 
+        /// <summary>
+        /// Encapsulates a python method with a <see cref="System.Func{T1, T2}"/>
+        /// </summary>
+        /// <typeparam name="T1">The data type</typeparam>
+        /// <typeparam name="T2">The output type</typeparam>
+        /// <param name="pyObject">The python method</param>
+        /// <returns>A <see cref="System.Func{T1, T2}"/> that encapsulates the python method</returns>
+        public static Func<T1, T2> ToFunc<T1, T2>(PyObject pyObject)
+        {
             using (Py.GIL())
             {
                 if (!pyObject.IsCallable()) return null;
-                dynamic toFunc = PythonEngine.ModuleFromString("x", testMod).GetAttr("to_func");
-                return toFunc(pyObject, typeof(T), typeof(TSecond)).AsManagedObject(typeof(Func<T, TSecond>));
+                dynamic method = GetModule().GetAttr("to_func");
+                return method(pyObject, typeof(T1), typeof(T2)).AsManagedObject(typeof(Func<T1, T2>));
+            }
+        }
+
+        /// <summary>
+        /// Try to get the length of arguments of a method
+        /// </summary>
+        /// <param name="pyObject">Object representing a method</param>
+        /// <param name="length">Lenght of arguments</param>
+        /// <returns>True if pyObject is a method</returns>
+        private static bool TryGetArgLength(PyObject pyObject, out int length)
+        {
+            using (Py.GIL())
+            {
+                dynamic inspect = Py.Import("inspect");
+
+                if (inspect.isfunction(pyObject))
+                {
+                    var args = inspect.getargspec(pyObject).args;
+                    length = new PyList(args).Length();
+                    return true;
+                }
+
+                if (inspect.ismethod(pyObject))
+                {
+                    var args = inspect.getargspec(pyObject).args;
+                    length = new PyList(args).Length() - 1;
+                    return true;
+                }
             }
+            length = 0;
+            return false;
+        }
+
+        /// <summary>
+        /// Creates a python module with utils methods 
+        /// </summary>
+        /// <returns>PyObject with a python module</returns>
+        private static PyObject GetModule()
+        {
+            return PythonEngine.ModuleFromString("x",
+                "from clr import AddReference\n" +
+                "AddReference(\"System\")\n" +
+                "from System import Action, Func\n" +
+                "def to_action1(pyobject, t1):\n" +
+                "    return Action[t1](pyobject)\n" +
+                "def to_action2(pyobject, t1, t2):\n" +
+                "    return Action[t1, t2](pyobject)\n" +
+                "def to_func(pyobject, t1, t2):\n" +
+                "    return Func[t1, t2](pyobject)");
```

Comment:
again, pretty awesome -- no action necessary here -- I'm typically partial to using C#'s verbatim string syntax (prefix string with `@`) so I can omit the `\n` characters... I suppose one concern, though, might be embedded `\n\r` windows style line endings, not sure if it would matter to python, but just a thought:
```csharp
        private static PyObject GetModule()
        {
            return PythonEngine.ModuleFromString("x", @"
from clr import AddReference
AddReference(""System"")
from System import Action, Func
def to_action1(pyobject, t1):
    return Action[t1](pyobject)
def to_action2(pyobject, t1, t2):
    return Action[t1, t2](pyobject)
def to_func(pyobject, t1, t2):
    return Func[t1, t2](pyobject)"
            );
        }
```

Also, R# can convert these for you. You can convert the concatenation to a constant using `Compute constant value` and then use `To verbatim string`.